java-manta-it/src/test/java/com/joyent/manta/http/MantaHttpHeadersIT.java

```diff
@@ -270,4 +272,36 @@ public void canSetDurability() throws IOException {
         Assert.assertEquals(actualHeaders.getDurabilityLevel().intValue(), durability,
                 "Durability not set to value on put");
     }
+
+    public void canFailToDeleteDirectoryOnBadIfMatch() throws IOException {
```

Comment:
Who uses this? As far as I can tell, there are no consumers.

Reply:
I'm not sure if you're asking about the `canFailToDeleteDirectoryOnBadIfMatch` method or the usage of `if-match`? JUnit/TestNG executes all public methods in a test class. The way it's designated as a test class is because it's referenced by [this line](https://github.com/joyent/java-manta/blob/master/java-manta-it/src/test/resources/testng-it.xml#L26) in the testng.xml configuration.

(`MantaClientTest#deleteWithHeadersPassesAlongHeaders` and the methods added to `StandardHttpHelperTest` are picked up the same way but with a different testng.xml)

Reply:
Ah, I see. I was wondering how the method gets called. I didn't realize that JUnit would take care of discovering and executing it, so this makes more sense to me now.